OpenSim/Sandbox/ExampleHopperDevice/exampleHopperDevice.cpp

```diff
@@ -1,428 +1,285 @@
+Model buildTestbed();   //defined in helperMethods.h
+Device* buildDevice();  //defined in buildDevice.cpp
 
-    // Get the ground frame which must be the base of any mechanism
-    auto& ground = testBed.getGround();
 
-    // Create a load of mass LOAD kg.
-    auto load = new OpenSim::Body("load", LOAD, Vec3(0), Inertia(1));
-    // Set properties of a sphere geometry to be used for the load.
-    OpenSim::Sphere sphere;
-    sphere.setFrameName("load");
-    sphere.set_radius(0.02);
-    sphere.setOpacity(0.5);
-    sphere.setColor(Vec3{ 0, 0, 1 });
-    load->addGeometry(sphere);
-    testBed.addBody(load);
+//------------------------------------------------------------------------------
+// Attach the device to any two PhysicalFrames in a model.
+// [Step 2, Task D]
+//------------------------------------------------------------------------------
+void connectDeviceToModel(OpenSim::Device& device, OpenSim::Model& model,
+    const std::string& modelFrameAname, const std::string& modelFrameBname)
+{
+    //TODO: Get references to the "anchor" joints in the device.
+
+    //TODO: Recall that the child frame of each anchor (WeldJoint) was attached
+    //      to the corresponding cuff. We will now attach the parent frames of
+    //      the anchors to modelFrameA and modelFrameB. First, get a reference
+    //      to the specified PhysicalFrames in model, then connect them to the
+    //      parent frames of each anchor (2 lines of code for each cuff).
+
+    // Add the device to the model. We need to add the device using
+    // addModelComponent() rather than addComponent() because of a bug in
+    // Model::initSystem().
+    model.addModelComponent(&device);
+
+    // Configure the device to wrap over the patella (if one exists). We are
+    // currently using a helper method because of a bug in GeometryPath.
+    addPathWrapHelper(model, "/device/cableAtoB", "patella", "thigh");
+}
 
-    auto grndToLoad = new OpenSim::FreeJoint("grndToLoad", ground, *load);
-    // Set the location of the load to (1, 0, 0).
-    grndToLoad->getCoordinateSet()[3].setDefaultValue(1.0);
-    testBed.addJoint(grndToLoad);
 
-    auto spring = new OpenSim::PointToPointSpring(
-        testBed.getGround(), Vec3(0),// point 1's frame and location in that frame
-        *load, Vec3(0),              // point 2's frame and location in that frame
-        SPRINGSTIFF, 1.0);           // spring stiffness and rest-length
+//------------------------------------------------------------------------------
+// Add a ConsoleReporter to the hopper model to display variables of interest.
+// [Step 1, Task A]
+//------------------------------------------------------------------------------
+void addConsoleReporterToHopper(Model& hopper)
+{
+    //TODO: Create a new ConsoleReporter. Set its name and reporting interval.
 
-    testBed.addForce(spring);
+    //TODO: Connect outputs from the hopper to the reporter's inputs.
```

Comment:
Was not clear to me which outputs to add (at first pass I only added the hopper height).